packages/sdk-auth/src/tokenProvider.js

```
@@ -91,6 +93,8 @@ export default class TokenProvider {
     return this.refreshTokenFlowPromise.then((refreshTokenInfo) => {
       this.refreshTokenFlowPromise = null
       return refreshTokenInfo
+    }).catch(() => {
+        this.refreshTokenFlowPromise = null
```

```suggestion
    }).catch((e) => {
        this.fetchTokenInfoPromise = null
        throw e
```
Just swallowing the error looks dangerous to me, how about above?

---

C&P error as your suggestion resets the wrong promise

---

Didn't realize and even that resetting the promise so updated the code.

---

Interesting, what does C&P error mean, @jenschude?

---

Copy&Paste. I meant your suggestion proposes instead of refreshTokenPromise a fetchTokenPromise

---

Ooops. Good one 👍